### PR TITLE
Whois sanitise domain

### PR DESCRIPTION
**Pull request by phily245**
Sanitises domains passed in here.

### CODE REVIEW

File: tests/Domains/WhoisTest.php [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+     * @dataProvider domainsToSanitise
+     */
+    public function testSanitisesDomains($unsanitised, $sanitised)
+    {
+        $this->assertEquals($sanitised, (new WhoisClient())->sanitiseDomain($unsanitised));
+    }
+
+    public function domainsToSanitise()
+    {
+        return [
```
Comment (rslsystems) [inline tests/Domains/WhoisTest.php:21]: Maybe add some more tests;  

- trailing slace: `http://foo.bar/`  
- whitespace on end: `http://foo.bar `  
- both the above `http://foo.bar/ `

Optional above with https as well

File: tests/Domains/WhoisTest.php [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+                'unsanitised' => 'foo.bar',
+                'sanitised' => 'foo.bar',
+            ],
+            [
+                'unsanitised' => 'foo.bar .baz',
+                'sanitised' => 'foo.bar%20.baz',
+            ],
+        ];
+    }
+}
```
Comment (phily245) [inline tests/Domains/WhoisTest.php:48]:
```suggestion
}

```

File: tests/Domains/WhoisTest.php (hunk 1 above)
Comment (phily245) [reply tests/Domains/WhoisTest.php:21]: Done, thanks :)

**Review by rslsystems — APPROVED** [review]
All good

**Review by rslsystems — APPROVED** [review]
Approving

**Pull request merged**